docs/testplan/srv6/SRv6-phoenixwing-ptf-testplan.md

```diff
@@ -0,0 +1,271 @@
+                                                        VM0105-t2
+```
+
+#### OVS_LINKs
+OVS_LINKs is a set of links which also has a connection with the PTF docker. This type of links could be used for running some PTF traffic verification. In this test plan, we use these links for verifying SRv6 encapsulated packets. "vlans" in the following configuration represent the ports connected with in ptf docker. This usage is the same as DUT's vlans usage in the topology file.
+
+Note: we don't want to add all links as OVS links, due to the following two reasons
+1. Limited ports in PTF docker
+2. We observe more chances to get ovs-switch crashed when the number of ovs flows used goes up.
+
+```
+  OVS_LINKs:
+    OVS6:
+      vlans:
+        - 39
+      start_vm_offset: 4
+      start_vm_port_idx: 3
+      end_vm_offset: 2
+      end_vm_port_idx: 1
+    OVS7:
+      vlans:
+        - 40
+      start_vm_offset: 5
+      start_vm_port_idx: 3
+      end_vm_offset: 2
+      end_vm_port_idx: 3
+```
+We could use "ovs-vsctl show" to find out ovs bridges first.
+```
+ubuntu@ubuntu:~$ sudo ovs-vsctl show
+c4e06a39-342c-489b-be81-09fbad6c739c
+    Bridge br_ovs6
+        Port VM0104-t3
+            Interface VM0104-t3
+        Port br_ovs6
+            Interface br_ovs6
+                type: internal
+        Port VM0102-t1
+            Interface VM0102-t1
+        Port inje-vms9-1-39
+            Interface inje-vms9-1-39
+...
+```
+Then we could use "ovs-ofctl dump-flows <bridge-name> to check flows and counters.
+```
+ubuntu@ubuntu:~$ sudo ovs-ofctl dump-flows br_ovs6
+ cookie=0x0, duration=7718.212s, table=0, n_packets=522, n_bytes=93445, in_port="VM0102-t1" actions=output:"VM0104-t3"
+ cookie=0x0, duration=7717.895s, table=0, n_packets=257, n_bytes=68105, priority=3,in_port="VM0104-t3" actions=output:"VM0102-t1",output:"inje-vms9-1-39"
+ cookie=0x0, duration=7717.812s, table=0, n_packets=4, n_bytes=280, in_port="inje-vms9-1-39" actions=output:"VM0104-t3"
+ cookie=0x0, duration=7718.202s, table=0, n_packets=0, n_bytes=0, priority=10,tcp,in_port="VM0104-t3",tp_src=179 actions=output:"VM0102-t1",output:"inje-vms9-1-39"
+ cookie=0x0, duration=7718.168s, table=0, n_packets=0, n_bytes=0, priority=10,tcp,in_port="VM0104-t3",tp_src=22 actions=output:"VM0102-t1",output:"inje-vms9-1-39"
+ cookie=0x0, duration=7718.158s, table=0, n_packets=264, n_bytes=25384, priority=10,tcp6,in_port="VM0104-t3",tp_src=179 actions=output:"VM0102-t1",output:"inje-vms9-1-39"
+ 
+ ...
+```
+### Setup configuration files
+Instead of using mininet xml, we want to use configdb json files to apply initial underlay configuraiton to each device. dut_cfg_file_loc is used in the topology file to provide the path for DUT's configdb json file and use cfg_file_loc under each VM to provide each VM's configdb json file. 
+
+```
+dut_cfg_file_loc : ../tests/srv6/7nodes_cisco/P1.json
+sonic_password_set : "admin"
+
+configuration:
+  PE1:
+    cfg_file_loc : ../tests/srv6/7nodes_cisco/PE1.json
```

Comment:
The topology file is supposed to contain logical relationship information between testbed components.
`dut_cfg_file_loc` and `cfg_file_loc` are detailed information for generating configuration files for testbed components. It is not appropriate to include them in topology file. It would be better to track these information in separate variable files.
The `deploy-mg` code should be updated to get these information from separate variable files.

Reply:
The DUT configuration is applied at deploy-mg, but other nodes configurations are done via add-topo. Since we don't use minigraph and want to use config_db.json to apply configuration, we copy over provided config files at the location when the configuration would be applied.